karpenter.tf

```diff
@@ -0,0 +1,100 @@
+locals {
+  # Karpenter Provisioners Config
+  # Use default var
+  karpenter_nodepools = var.karpenter_nodepools
+
+  # Karpenter Nodetemplate Config
+  karpenter_nodeclasses = coalescelist(var.karpenter_nodeclasses, [
+    {
+      nodeclass_name = "default"
+      karpenter_subnet_selector_maps = [{
+        tags = {
+          "Name" = "aft-app-ap-southeast*"
+        },
+        }
```

Comment:
We shouldn't use this, better if we can use karpenter discovery tag.

`karpenter.sh/discovery: "${CLUSTER_NAME}"`

But this means, we need to allow adding subnet tag with a custom value using `aws_ec2_tag` and allow SCP. OR

we can use existing tags. ex:

`"kubernetes.io/role/internal-elb" = "1"`

Reply:
introduce `karpenter_subnet_selector_tags` variable to pass both key and value of the tags 
```
karpenter_subnet_selector_tags = {
  "Name" = "App*"
}
```